### ISSUE TITLE
Unique constraint shouldn't treat null-values as equal

### ISSUE THREAD

**Issue by slavafomin** [balderdashy/sails-disk#46]
Hello!

I have a model with `email` field with `unique` constraint. However, I can't have two entities with `email = null` cause module treat it as non-unique. Both MySQL and PostgreSQL treat NULL values as not-equal and it makes great sense.

I hope this is the correct repository to post issue to. BTW I'm using Sails.js.

What do you think? Thank you!

---

Here's the error I'm getting:

```json
{
  "error": "E_VALIDATION",
  "status": 400,
  "summary": "1 attribute is invalid",
  "invalidAttributes": {
    "email": [
      {
        "value": null,
        "rule": "unique",
        "message": "A record with that `email` already exists (`null`)."
      }
    ]
  }
}
```

**Comment by sailsbot** [comment]
Thanks for posting, @slavafomin.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
